resources/assets/js/components/AcceptPayment.vue

```diff
@@ -0,0 +1,170 @@
+                <li>Follow the prompts to complete the transaction.</li>
+                <li>It is not required to provide a paper receipt for swiped card transactions.</li>
+              </ol>
+            </template>
+            <template v-else-if="payment.method == 'squarecash'">
+              <p class="font-weight-bold">Check with the treasurer before marking SquareCash payments as complete.</p>
+              <ol>
+                <li>View the email saying that the money has been deposited into our account. This is the second email of a given transaction.</li>
+                <li>It is not required to provide a paper receipt for SquareCash transactions.</li>
+              </ol>
+            </template>
+          </div>
+        </div>
+
+        <div class="row">
+          <div class="col-12 col-md-6">
+            <button type="submit" class="btn btn-primary float-right">Submit</button>
+          </div>
+        </div>
+
+      </form>
+    </div>
+  </div>
+</template>
+
+<script>
+  /*
+   *  @props transactionType: Morph of the Transaction model to attach the payment
+   *  @props transactionID: ID of the specific morph of a given Transaction Model
+   *  @props amount: The amount of currency in this payment
+   *  @emit done: Event emitted after a payment is recorded successfully
+   */
+
+  import { numeric, required, sameAs } from 'vuelidate/lib/validators';
+
+  export default {
+    props: {
+      transactionType: {
+        type: String,
+        required: true
+      },
+      transactionId: {
+        type: Number,
+        required: true
+      },
+      amount: {
+        required: true
+      }
+    },
+    data() {
+      return {
+        payment: {
+          payable_id: this.transactionId,
+          payable_type: "App\\" + this.transactionType,
+          method: "",
+          amount: null
+        },
+        paymentMethods: [
+          {value: "cash", text: "Cash"},
+          {value: "check", text: "Check"},
+          {value: "swipe", text: "Swiped Card"},
+          {value: "squarecash", text: "SquareCash"},
+        ],
+        baseUrl: "/api/v1/payments",
+      }
+    },
+    methods: {
+      submit: function() {
+        axios.post(this.baseUrl, this.payment)
+          .then(response => {
+            this.$emit('done');
+          })
+          .catch(response => {
+            console.log(response);
+            sweetAlert("Connection Error", "Unable to record payment. Check your internet connection or try refreshing the page.", "error");
+          })
+      }
+    },
+    computed: {
+      expectedAmount: function() {
+        if (this.payment.method == 'swipe') {
+          return parseFloat(this.amount) + 3;
+        } else {
+          return parseFloat(this.amount);
+        }
+      }
+    },
+    validations: {
+      payment: {
+        method: {
+          required
+        },
+        amount: {
```

Comment:
I _think_ this should be checking for discrepancies in expected vs paid amount if I'm reading this right. If so, it's not working. I recorded a $99 payment for a $10 package.

Reply:
I forgot the code that prevents submit if it's wrong. The validation is working as intended.